### DISCUSSION TITLE
Remapping coordinates with a function

### DISCUSSION THREAD

**Discussion by mgunyho** [pydata/xarray#5222] [Q&A]
Hello,

I'm frequently using something like
```py
da = da.assign_coords(x=some_function(da.x))
```
to reassign the coordinates using a callable. However, usually, I have more descriptive names than `da` and `x`, so I end up having to type a lot:
```py
some_array = some_array.assign_coords(dimension_with_a_long_name=some_function(some_array.dimension_with_a_long_name))
```
I was hoping that there could be a way to remap the coordinates using a function, without having to retype the array and coordinate name, like
```py
some_array = some_array.map_coords(dimension_with_a_long_name=some_function)
```

I see that it's possible to pass a Callable to `assign_coords`, but that is computed on the whole array (`da` in the first example) and not on the coordinate. Have I missed something, is something like this possible with the current API? Or should I raise an issue about this?

**Comment by max-sixty** [comment]
Thanks for starting a discussion @mgunyho 

My initial sense is this may be too specific to add a method for (though people should 👍  if they have the same issue). But I wonder whether we could make this easier using the more general methods — e.g. `da.coords = da.coords.map(func)`?

**Reply by mgunyho** [reply]
Yeah, a separate method is probably too much. Another option I thought of could be some `dim` kwarg to `assign_coords`, but that might interfere with more complicated use cases.

I just realized that I can do `da["dim_name"] = func(da.dim_name)`, which still requires typing the coordinate name twice, but at least it can be auto-completed.
